internal/state/dataplane/configuration.go

```diff
@@ -83,13 +83,43 @@ type MatchRule struct {
 	// the entire resource.
 	Source *v1beta1.HTTPRoute
 	// BackendGroup is the group of Backends that the rule routes to.
-	BackendGroup graph.BackendGroup
+	BackendGroup BackendGroup
 	// MatchIdx is the index of the rule in the Rule.Matches.
 	MatchIdx int
 	// RuleIdx is the index of the corresponding rule in the HTTPRoute.
 	RuleIdx int
 }
 
+// BackendGroup represents a group of Backends for a routing rule in an HTTPRoute.
+type BackendGroup struct {
+	// Source is the NamespacedName of the HTTPRoute the group belongs to.
+	Source types.NamespacedName
+	// Backends is a list of Backends in the Group.
+	// If there are no backends defined for the routing rule, this field will be nil.
```

Comment:
is there any particular reason why we mention Backends will be `nil`? If there are no backends - nil value seems natural in any case